src/utils/avail.rs

```diff
@@ -0,0 +1,86 @@
+use std::process::Stdio;
+use std::string::FromUtf8Error;
+use std::{fs, io};
+
+use serde_json::{json, Value};
+use thiserror::Error;
+
+use crate::app::config::DALayer;
+use crate::utils::cmd::{execute_cmd, execute_cmd_stdout};
+use crate::utils::paths::{get_app_home, get_madara_home};
+
+#[derive(Debug, Error)]
+pub enum AvailKeyGenError {
+    #[error("Failed to read file: {0}")]
+    FailedToIoFilesystem(#[from] io::Error),
+    #[error("Failed to parse output: {0}")]
+    FailedToParseOutput(#[from] FromUtf8Error),
+    #[error("Failed to parse to json: {0}")]
+    FailedToParseToJson(#[from] serde_json::Error),
+}
+
+pub fn setup_and_generate_keypair(app_chain: &str) -> Result<DALayer, AvailKeyGenError> {
+    install_subkey()?;
+    let avail = generate_keypair(app_chain)?;
+
+    Ok(avail)
+}
+fn install_subkey() -> Result<(), AvailKeyGenError> {
```

Comment:
instead of installing the binary, let's import the package directly in our Cargo.toml and call the function needed to generate the keypair